deepfence_worker/ingesters/cloud_resource.go

```diff
@@ -131,6 +144,24 @@ func CommitFuncCloudResource(ns string, cs []CloudResource) error {
 			"shown_types": TopologyCloudResourceTypes,
 		},
 	)
+
+	if len(hosts) > 0 {
+		_, err = tx.Run(`
+		UNWIND $batch as row
+		WITH row
+		OPTIONAL MATCH (n:Node{node_id:row.node_id})
+		WITH n, row as row
+		WHERE n IS NULL or n.active=false
+		MERGE (m:Node{node_id:row.node_id})
+		MERGE (cp:CloudProvider{node_id:row.cloud_provider})
+		MERGE (cr:CloudRegion{node_id:row.cloud_region})
+		MERGE (cp) -[:HOSTS]-> (cr)
+		MERGE (cr) -[:HOSTS]-> (m)
+		SET m+=row, m.updated_at = TIMESTAMP(), cp.active = true, cp.pseudo = false, cr.active = true`,
+			map[string]interface{}{"batch": hosts},
+		)
+	}
+
```

Comment:
Maybe not for this PR but we need to move this piece of code here instead:
https://github.com/deepfence/ThreatMapper/blob/v2/deepfence_worker/cronjobs/neo4j.go:356

This is purely for performance reason, accessing `CloudProvider` and `CloudRegion` will result in deadlock situations under heavy load